src/core/core.tooltip.js

```diff
@@ -743,13 +755,36 @@ var exports = module.exports = Element.extend({
 
 			var titleFontSize = vm.titleFontSize;
 			var titleSpacing = vm.titleSpacing;
+			var drawColorBoxes;
+			var xLinePadding = 0;
+			if (this._options.mode === 'dataset') {
+				drawColorBoxes = vm.displayColors;
+				xLinePadding = drawColorBoxes ? (titleFontSize + 2) : 0;
+			}
 
 			ctx.fillStyle = mergeOpacity(vm.titleFontColor, opacity);
 			ctx.font = helpers.fontString(titleFontSize, vm._titleFontStyle, vm._titleFontFamily);
 
 			var i, len;
 			for (i = 0, len = title.length; i < len; ++i) {
-				ctx.fillText(title[i], pt.x, pt.y);
+				var textColor = mergeOpacity(vm.labelTextColors[i], opacity);
+				// Draw Legend-like boxes if needed
+				if (drawColorBoxes) {
```

Comment:
This looks like copy/pasted code from line 829. Please refactor out a shared function

Reply:
i refactor out a helper function